app/Models/Campaign.php

```diff
@@ -50,4 +50,9 @@ public function adventure()
     {
         return $this->belongsTo(\App\Models\Adventure::class);
     }
+
+    public function generateCode()
+    {
+        return dechex(microtime(true)*1000);
```

Comment:
```suggestion
        return dechex(microtime(true)*10000);
```
after i did some testing locally, microtime actually returns a number with 4 decimal places, so lets use all of them :)